fontbe/src/features.rs

```diff
@@ -78,26 +93,46 @@ impl FeatureWork {
             .build_raw(&glyph_order, Default::default())
             .map_err(|_| {
                 Error::FeaError(format!(
-                    "{:?} build_raw failed; no useful diagnostic available",
-                    fea_file
+                    "{} build_raw failed; no useful diagnostic available",
+                    feature_source
                 ))
             })?;
         Ok(font)
     }
+
+    fn compile_memory(
+        &self,
+        fea_content: &String,
+        glyph_order: GlyphMap,
+    ) -> Result<FontBuilder, Error> {
+        // Will you not parse?!
+
+        // TODO write out the feature content on failure
```

Comment:
you already have code to write to debug_file when compile fails, you may as well factor that out to its own function and also call it when parsing fails

Reply:
Oh good catch, fixed